src/array.rs

```diff
@@ -16,19 +16,31 @@ impl<T> Default for Array<T> {
 }
 
 impl<T: RuntimeType> Array<T> {
+    /// Creates an empty array.
     pub fn new() -> Self {
         Self::default()
     }
 
-    pub fn clear(&mut self) {
-        // TODO: drop members, CoTastkMemFree, zero members
+    /// Returns `true` if the array is empty.
+    pub fn is_empty(&self) -> bool {
+        self.len == 0
     }
 
-    pub fn as_slice(&self) -> &[T] {
-        if self.data.is_null() {
-            return &[];
+    /// Returns the length of the array.
+    pub fn len(&self) -> usize {
+        self.len as usize
+    }
+
+    /// Clears the contents of the array.
+    pub fn clear(&mut self) {
+        if self.is_empty() {
+            return;
+        }
+
+        unsafe {
+            std::ptr::drop_in_place(&mut self[..]);
```

Comment:
I think this is not fully correct. If we call `clear` and one of the values in the array panics during its drop, we'll unwind the stack and call `clear` again in the drop impl of Array leading us right back here to the where the array elements drop impls will be called again (double free). 

I think we have to first swap the pointers and then drop. If one of the elements panics, we may leak memory, but we won't double free.

Reply:
Fixed.